### PR TITLE
deferred read

### PR DESCRIPTION
**Pull request by dlovell**

### CODE REVIEW

File: examples/deferred_read_csv.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import letsql as ls
+from letsql.common.utils.defer_utils import (
+    deferred_read_csv,
```
Comment (mesejo) [inline examples/deferred_read_csv.py:3]: Should we add `deferred_read_csv` as an api function?

Comment (dlovell) [reply examples/deferred_read_csv.py:3]: we can, though i would like it to be `read_csv` with "deferred" implied

related, we should not introspect on backends to detect what the csv/parquet source is and instead require users to use `deferred_read_{csv,parquet}` (or eventual api function `read_{csv,parquet}`) if they want to be able to cache the expr

File: python/letsql/backends/let/hotfix.py [hunk 2]
```diff
@@ -126,7 +127,12 @@ def exists(self):
 def _letsql_find_backend(self, *, use_default=True):
     # FIXME: push this into LETSQLAccessor
     try:
-        current_backend = self._find_backend._original(self, use_default=use_default)
+        if tuple(op.source for op in self.op().find(Read)):
+            current_backend = letsql.options.backend
```
Comment (mesejo) [inline python/letsql/backends/let/hotfix.py:131]: I think `letsql.options.backend` may be null if it has not been initialized; in general, it is safer to do:
```python
    from letsql.config import _backend_init
    current_backend = _backend_init() 
```

File: python/letsql/common/utils/dask_normalize_expr.py [hunk 3]
```diff
@@ -167,6 +171,44 @@ def normalize_module(module):
...
+def normalize_read(read):
+    path = dict(read.read_kwargs).get("path") or dict(read.read_kwargs).get("source")
+    if isinstance(path, (str, pathlib.Path)):
+        path = str(path)
+        if path.startswith("http") or path.startswith("https:"):
+            import requests
+
+            resp = requests.head(path)
+            resp.raise_for_status()
+            dct = {
```
Comment (mesejo) [inline python/letsql/common/utils/dask_normalize_expr.py:184]: Is the variable `dct` a dictionary?

**Review by mesejo — APPROVED** [review]

File: python/letsql/common/utils/dask_normalize_expr.py (hunk 3 above)
Comment (dlovell) [reply python/letsql/common/utils/dask_normalize_expr.py:184]: yes

Comment (dlovell) [reply python/letsql/common/utils/dask_normalize_expr.py:184]: rather, it should be (thought doesn't need to be)

**Pull request merged**